build-logic/java-api-extractor/src/main/java/org/gradle/internal/tools/api/impl/JavaApiMemberWriter.java

```diff
@@ -108,6 +109,22 @@ public void writeMethod(/* Nullable */ InnerClassMember declaringInnerClass, Met
         mv.visitEnd();
     }
 
+    private static Optional<Integer> calculateNonAnnotableParameterCount(ClassMember classMember, /* Nullable */ InnerClassMember declaringInnerClass, MethodMember method) {
+        if (method.getName().equals("<init>")) {
+            if ((classMember.getAccess() & ACC_ENUM) == ACC_ENUM) {
+                // Enum constructors have an implicit String and int parameter containing
+                // the name and ordinal of the value that is non-annotable.
+                return Optional.of(2);
```

Comment:
💭 This is the main change in the PR: we are now also handling enum constructors.